integration-test/src/main/java/com/sequenceiq/it/cloudbreak/util/wait/WaitUtil.java

```diff
@@ -449,13 +451,37 @@ private boolean checkSdxInstanceState(SdxClient sdxClient, String sdxName, Strin
         List<InstanceGroupV4Response> instanceGroups = sdxClient.getSdxClient().sdxEndpoint().getDetail(sdxName, Set.of())
                 .getStackV4Response()
                 .getInstanceGroups();
-        return checkInstanceState(hostGroup, desiredState, instanceGroups);
+        SdxClusterResponse sdxResponse = sdxClient.getSdxClient().sdxEndpoint().get(sdxName);
```

Comment:
This method and checkDistroxInstanceState are almost identical. I would try to refactor them to have one method. I think it is only the log and exception messages that differ within method body.